crates/egui/src/widgets/slider.rs

```diff
@@ -213,6 +218,30 @@ impl<'a> Slider<'a> {
         self
     }
 
+    /// Set specific points for the slider to snap to. These points are visualized with small dots.
+    /// This only has an effect when [`smart_aim`] is ON. `snap_radius` defines the range around
+    /// the mouse where the slider will snap to the given values. To be useful, this should be
+    /// somewhat large (and definitely larger than [`InputState::aim_radius`]). Default is 0.1.
+    pub fn smart_aim_values<Num: emath::Numeric>(
+        mut self,
+        smart_aim_values: Vec<Num>,
+        snap_radius: f32,
+    ) -> Self {
+        self.snap_radius = snap_radius.clamp(0.0, 1.0);
+        self.smart_aim_values = smart_aim_values
+            .into_iter()
+            .map(|n| n.to_f64().clamp(*self.range.start(), *self.range.end()))
+            .collect();
+        self.smart_aim_values
+            .sort_by(|a, b| match a.partial_cmp(b) {
+                Some(ordering) => ordering,
+
+                // NaNs are handled silently
+                None => Ordering::Greater,
```

Comment:
this is unsound. This can make `a < b` and `b < a` for some values of a,b, which could theoretically make the sort hang.